runtime/jcl/common/jclcinit.c

```diff
@@ -712,6 +717,35 @@ initializeRequiredClasses(J9VMThread *vmThread, char* dllName)
 		}
 	}
 
+	if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_LOAD_AGENT_MODULE)
+			&& (NULL != vm->modulesPathEntry->extraInfo)) {
+		Trc_JCL_initializeRequiredClasses_addAgentModuleEntry(vmThread);
+		const char *module = vm->jimageIntf->jimagePackageToModule(
+				vm->jimageIntf, (UDATA) vm->modulesPathEntry->extraInfo,
+				"jdk/internal/agent");
+		if (NULL != module) {
+			J9VMSystemProperty *systemProperty = NULL;
+			/* Handle the case where there are no user-specified modules. In this case, there
+			 * is typically one system property but no user-set "add-modules" arguments.
+			 */
+			if ((0 == vm->addModulesCount) &&
+					(J9SYSPROP_ERROR_NONE == vm->internalVMFunctions->getSystemProperty(vm, ADDMODS_PROPERTY_BASE "0", &systemProperty))
+					) {
```

Comment:
{ on own line

Reply:
Also not a fan of this indent style for the multi-line if. Not sure I have a better suggestion offhand, other than spacing in to match the if.

Reply:
The indenting is done by the auto-formatter.  I reduced the number of tabs, though.